source/server/guarddog_impl.cc

```diff
@@ -227,8 +229,9 @@ void GuardDogImpl::invokeGuardDogActions(
 }
 
 GuardDogImpl::WatchedDog::WatchedDog(Stats::Scope& stats_scope, const std::string& thread_name,
-                                     const WatchDogSharedPtr& watch_dog)
+                                     const WatchDogImplSharedPtr& watch_dog)
     : dog_(watch_dog),
+      last_touch_count_(dog_->touchCount()),
```

Comment:
Perhaps you should set this to 0 (the last touch count)

otherwise we'll fall through `if (watched_dog->last_touch_count_ != touch_count)` on the first iteration of the new watchdog in `step`. We'll end up using `const auto ltt = watched_dog->last_touch_time_;` without it being explicitly set

Reply:
Notice line 179-180 above, where the WatchedDog is created and touched in the following line to guarantee that the first guarddog step after creation goes down the "watched_dog->last_touch_count_ != touch_count" branch.